src/Orchard.Cms.Web/Modules/Orchard.Recipes/RecipeSteps/RecipesStep.cs

```diff
@@ -35,49 +31,38 @@ public class RecipesStep : RecipeExecutionStep
             ]
          }
         */
-        public override async Task ExecuteAsync(RecipeExecutionContext context)
+        public async Task ExecuteAsync(RecipeExecutionContext context)
         {
-            var step = context.RecipeStep.Step.ToObject<InternalStep>();
+            if (!String.Equals(context.Name, "Recipes", StringComparison.OrdinalIgnoreCase))
+            {
+                return;
+            }
+
+            var step = context.Step.ToObject<InternalStep>();
             var recipesDictionary = new Dictionary<string, IDictionary<string, RecipeDescriptor>>();
 
             foreach (var recipe in step.Values)
             {
-                Logger.LogInformation("Executing recipe '{0}' in extension '{1}'.", recipe.Name, recipe.ExecutionId);
-
-                try
+                var recipes = recipesDictionary.ContainsKey(recipe.ExecutionId) ? recipesDictionary[recipe.ExecutionId] : default(IDictionary<string, RecipeDescriptor>);
```

Comment:
I agree, this is some really stupid code ;) oh, wait, it's yours 😄 I just moved it. I will fix.